lms/djangoapps/certificates/api.py

```diff
@@ -425,51 +396,66 @@ def example_certificates_status(course_key):
     return ExampleCertificateSet.latest_status(course_key)
 
 
+def _safe_course_key(course_key):
+    if not isinstance(course_key, CourseKey):
+        try:
+            return CourseKey.from_string(course_key)
+        except InvalidKeyError:
+            log.warning('Unable to parse course_key "%s"', course_key, exc_info=True)
+            return
+    return course_key
+
+
+def _course_from_key(course_key):
+    try:
+        return CourseOverview.get_from_id(_safe_course_key(course_key))
+    except:  # pylint: disable=bare-except
```

Comment:
This is another manifestation of the scary mask-exception pattern.
Also, please see "Don't Return Null" on p.110 of the Clean Code Book.

Reply:
Ditto above.

Reply:
Also, it looks like `get_from_id()` will return `None` if the course doesn't exist: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/content/course_overviews/models.py#L264-L301

(And even better, the comment "Raises: CourseOverview.DoesNotExist if the course specified by course_id was not found" in that method is a lie.)

Reply:
hmm.. as I read it, `get_from_id` doesn't return `None`, but does raise `DoesNotExist` exception when the course doesn't exist in either SQL or modulestore.  See the last line in that method.

Reply:
hah, I missed the call to `load_from_module_store()`.